From 4be8163516fad9a698c6b1097f93991a8cb80266 Mon Sep 17 00:00:00 2001
From: Samuel Ortiz <sameo@linux.intel.com>
Date: Fri, 5 Sep 2014 00:15:06 +0200
Subject: [PATCH] input: ft5x06: Reset the chipset before doing the initial I/O

On the probe routine we're doing the initial register settings
through i2c, before physically resetting the chipset.
That works fine on a cold boot as it seems the BIOS puts the
chipset in D0. On a warm boot, the chipset may be in D3 and
trying to it through i2c then fails. That issue is observed
for example after a watchdog triggered reset.

Issue: IRDA-501
Change-Id: I8268a893e2393275cfa59b0ad9c035f4b5aef895
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
---
 drivers/input/touchscreen/ft5x06_ts.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/input/touchscreen/ft5x06_ts.c b/drivers/input/touchscreen/ft5x06_ts.c
index f27885cb..e6edd27 100644
--- a/drivers/input/touchscreen/ft5x06_ts.c
+++ b/drivers/input/touchscreen/ft5x06_ts.c
@@ -711,17 +711,19 @@ static int ft5x0x_ts_probe(struct i2c_client *client,
 		goto err_sysfs_create;
 	}
 #endif
-	err = ft5x0x_reg_init(ts_data);
+	err = ft5x0x_wake_device(ts_data);
 
 	if (err) {
-		dev_err(&client->dev, "reg init failed\n");
+		dev_err(&client->dev, "enable device failed\n");
 		goto err_sysfs_create;
 	}
 
-	err = ft5x0x_wake_device(ts_data);
+	dev_err(&client->dev, "Doing reg init now\n");
+
+	err = ft5x0x_reg_init(ts_data);
 
 	if (err) {
-		dev_err(&client->dev, "enable device failed\n");
+		dev_err(&client->dev, "reg init failed\n");
 		goto err_sysfs_create;
 	}
 
-- 
1.8.3.2

